dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/export/csv/TrackerCsvTrackedEntityServiceTest.java

```diff
@@ -0,0 +1,170 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import org.geotools.geojson.geom.GeometryJSON;
+import org.hisp.dhis.common.ValueType;
+import org.hisp.dhis.webapi.controller.tracker.view.Attribute;
+import org.hisp.dhis.webapi.controller.tracker.view.TrackedEntity;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.locationtech.jts.geom.Geometry;
+
+class TrackerCsvTrackedEntityServiceTest
+{
+
+    private TrackerCsvTrackedEntityService service;
+
+    private Instant instant;
+
+    @BeforeEach
+    void setUp()
+    {
+        service = new TrackerCsvTrackedEntityService();
+    }
+
+    private TrackedEntity createTrackedEntity()
+    {
+        TrackedEntity trackedEntity = new TrackedEntity();
+        trackedEntity.setTrackedEntity( "Test tracked entity" );
+        instant = Instant.parse( "2022-09-29T15:15:30.00Z" );
+        trackedEntity.setCreatedAt( instant );
+        trackedEntity.setOrgUnit( "Test org unit" );
+
+        return trackedEntity;
+    }
+
+    @Test
+    void singlePointCoordinatesEntityIsWritten()
+        throws IOException
+    {
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        TrackedEntity trackedEntity = createTrackedEntity();
+
+        GeometryJSON geometryJSON = new GeometryJSON();
+        String pointCoordinates = "[40,5]";
+        Geometry geometry = geometryJSON.read( "{\"type\":\"Point\", \"coordinates\": " + pointCoordinates + " }" );
+        trackedEntity.setGeometry( geometry );
+
+        service.writeEvents( out, List.of( trackedEntity ), false );
+        assertEquals( "\"Test tracked entity\",," + instant.toString()
+            + ",,,,\"Test org unit\",false,false,false,\"POINT (40 5)\",5.0,40.0,,,,,,,,,\n", out.toString() );
+    }
+
+    @Test
+    void singleLineCoordinatesEntityIsWritten()
+        throws IOException
+    {
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        TrackedEntity trackedEntity = createTrackedEntity();
+
+        GeometryJSON geometryJSON = new GeometryJSON();
+        String lineCoordinates = "[[40,5],[41,6]]";
+        Geometry geometry = geometryJSON.read( "{\"type\":\"LineString\", \"coordinates\": " + lineCoordinates + " }" );
+        trackedEntity.setGeometry( geometry );
+
+        service.writeEvents( out, List.of( trackedEntity ), false );
+        assertEquals( "\"Test tracked entity\",," + instant.toString()
+            + ",,,,\"Test org unit\",false,false,false,\"LINESTRING (40 5, 41 6)\",,,,,,,,,,,\n", out.toString() );
+    }
+
+    @Test
+    void multipleTrackedEntitiesAreWritten()
+        throws IOException
+    {
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        List<TrackedEntity> trackedEntities = new ArrayList<>();
+        IntStream.range( 0, 3 ).forEach( i -> trackedEntities.add( createTrackedEntity() ) );
+
+        assertEquals( 3, trackedEntities.size() );
```

Comment:
I don't think we need this assert here, you just created 3 TEIs, no need to check if that is true

Reply:
I know, I added this one to avoid possible NPE and have a cleaner error message in case there's a change in createTrackedEntity(). Do you think it's not necessary anyway?

Reply:
But no matter how createTrackedEntity() changes you will have 3 elements in the list.
I think here you are asserting your setup for the test instead of what you are testing, it can be confusing